meta-lmp-base/recipes-support/lmp-el2go-auto-register/lmp-el2go-auto-register/lmp-el2go-auto-register

```diff
@@ -19,6 +19,7 @@ PIN = os.environ.get("PKCS11_PIN", "87654321")
 SO_PIN = os.environ.get("PKCS11_SOPIN", "12345678")
 SOTA_DIR = os.environ.get("SOTA_DIR", "/var/sota")
 HANDLERS = os.environ.get("HANDLERS", "aws-iot,aktualizr-lite")
+PACMAN_TYPE = os.environ.get("PACMAN_TYPE", "ostree+compose_apps")
```

Comment:
This is fine, but a user would still have to make additional changes to bitbake to get this set, right?

We could probably do this automagically with something like:
```
from shutil import which
def _has_docker() -> bool:
    return which("docker") is not None
```
Then you can know at runtime what packagemanager to assign.

Reply:
Indeed, we should support users with docker and without necessarily wanting to use composeapp.

We can for sure reuse the composeapp packageconfig and make it a distro feature, having both registration related recipes using the same package config.

Since this is a script, the generation of default.env could be done at build time, otherwise we would need some sort of sed to have it hardcoded.

Reply:
k. Running composeapp+ostree doesn't require apps, but this could be theoretically possible. I guess what we need to avoid is allowing someone to ask for composeapp+ostree if docker isn't on the system. Maybe the recipe can detect that.

Reply:
> Still, a good approach could be to add a basic (empty) environment file in meta-lmp-base and In the .bb file, append the `PACMAN_TYPE` line to it. Then we can simply copy that basic file to meta-subscriber-overrides, add whatever variables to it, and set `FILESEXTRAPATH` in bbappend.

Yeah, we already provide an empty default.env that was created to facilitate the process to override it via meta-sub, as you suggested, so I think we can land the current PR (since it won't change the default behavior) and still allow users to create a custom default.env with a different PACMAN_TYPE.

If needed we can extend this in a more complete approach later.